### PR TITLE
feat(ai-function): create testset if not set testSetID

### PR DESCRIPTION
**Pull request by wangzhuzhen**
#### What this PR does / why we need it:
AI function create  testcase will create testset if not set testSetID

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog
Feature: Support ai function create testset if not set testSetID（支持AI function 在测试集 ID 未指定的情况下创建测试集.）

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Support ai function create testset if not set testSetID          |
| 🇨🇳 中文    |       支持AI function 在测试集 ID 未指定的情况下创建测试集.       |

#### Need cherry-pick to release versions?
/cherry-pick release/2.4

### CODE REVIEW

File: bundle/testset.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/erda-project/erda/pkg/http/httpserver"
+	"github.com/erda-project/erda/pkg/http/httputil"
+)
+
+func (b *Bundle) CreateTestSet(req apistructs.TestSetCreateRequest) (*apistructs.TestSet, error) {
+	host, err := b.urls.ErdaServer()
+	if err != nil {
+		return &apistructs.TestSet{}, err
+	}
+	var resp httpserver.Resp
```
Comment (sfwn) [inline bundle/testset.go:35]: Use defined struct directly, no need to marshal&unmarshal by yourself again.
Refer to: `apistructs.ProjectResourceResponse`.

File: bundle/testset.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func (b *Bundle) GetTestSets(req apistructs.TestSetListRequest) ([]apistructs.TestSet, error) {
+	host, err := b.urls.ErdaServer()
+	if err != nil {
+		return nil, err
+	}
+	pathVars := fmt.Sprintf("?recycled=%v&parentID=%d&projectID=%d", req.Recycled, *req.ParentID, *req.ProjectID)
+	var resp httpserver.Resp
+	r, err := b.hc.Get(host).Path("/api/testsets"+pathVars).
```
Comment (sfwn) [inline bundle/testset.go:71]: Let path clean. And use `.Params()` for query params.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: bundle/testset.go (hunk 1 above)
Comment (wangzhuzhen) [reply bundle/testset.go]: OK

File: bundle/testset.go (hunk 2 above)
Comment (wangzhuzhen) [reply bundle/testset.go]: OK

**Review by sfwn — APPROVED** [review]

**Pull request merged**